homeworks/hw12/src/bank.rs

```diff
@@ -0,0 +1,869 @@
+    /// Withdraws the specified amount from the account.
+    ///
+    /// # Arguments
+    ///
+    /// * `account` - The account to withdraw from.
+    /// * `amount` - The amount to withdraw from the account.
+    ///
+    /// # Returns
+    /// TransactionId for processing the transaction
+    /// BankError if the account does not exist or the amount is negative or insufficient funds
+    /// # Errors
+    /// AmountNegativeError
+    /// AccountNotFoundError
+    /// InsufficientFundsError
+    ///
+    /// ```
+
+    fn withdraw(&mut self, account: &str, amount: Money) -> Result<TransactionId, BankError>;
+    /// Transfers the specified amount from one account to another.
+    ///
+    /// # Arguments
+    ///
+    /// * `sender` - The name of the account from which the amount will be transferred.
+    /// * `receiver` - The name of the account to which the amount will be transferred.
+    /// * `amount` - The amount to transfer.
+    ///
+    /// # Errors
+    /// AmountNegativeError
+    /// AccountNotFoundError
+    /// InsufficientFundsError
+    /// SomeAccountTransferError
+    ///
+    /// Returns an error if either the sender or receiver account does not exist, or if
+    /// the sender account does not have sufficient balance to cover the transfer amount.
+    ///
+    /// ```
+    fn transfer(
+        &mut self,
+        sender_account: &str,
+        receiver_account: &str,
+        amount: Money,
+    ) -> Result<TransactionId, BankError>;
+    /// Returns the current balance of the account.
+    /// # Arguments
+    ///
+    /// * `account` - The name of the account for which to retrieve the transaction history.
+    ///
+    /// # Returns
+    /// The current balance of the account.
+    /// # Errors
+    /// AccountNotFoundError
+    /// ```
+    fn get_balance(&self, account: &str) -> Result<Money, BankError>;
+    /// Returns the transaction history of the specified account.
+    ///
+    /// # Arguments
+    ///
+    /// * `account` - The name of the account for which to retrieve the transaction history.
+    ///
+    /// # Returns
+    ///
+    /// A vector of strings representing the transaction history of the account.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the specified account does not exist.
+    /// ```
+    fn get_history(&self) -> Result<Vec<Operation>, BankError>;
+    /// Returns the transaction history of the specified account.
+    ///
+    /// # Arguments
+    ///
+    /// * `account` - The name of the account for which to retrieve the transaction history.
+    ///
+    /// # Returns
+    ///
+    /// A vector of strings representing the transaction history of the account.
+    ///
+    /// # Errors
+    /// BankError
+    /// ```
+
+    fn get_account_history(&self, account: &str) -> Result<Vec<Operation>, BankError>;
+    /// Replays the transaction history stored in a source_bank for the new Bank instance.
```

Comment:
```suggestion
    /// ```
    fn get_account_history(&self, account: &str) -> Result<Vec<Operation>, BankError>;

    /// Replays the transaction history stored in a source_bank for the new Bank instance.
```